scalarxmat44.c

```diff
@@ -1,107 +1,96 @@
 #include <immintrin.h>
+#include <math.h>
 #include <stdint.h>
 #include <stdio.h>
 #include <string.h>
-#include <sys/time.h>
+#include <time.h>
+#define N 1000000000   // loops
 
-#define N 4
-
-int
-main() {
-
-    // allocate memory for array
+void
+checkArraysEqual(float A[4][4], float B[4][4], float epsilon) {
     int are_equal = 1;
-    float lamda = 5.0;
-    int count = 0;
-    float(*A)[N] = malloc(sizeof(float[N][N]));
-    if (!A) {
-        fprintf(stderr, "error allocating memory for matrix\n");
-        exit(1);
-    }
-    float(*B)[N] = malloc(sizeof(float[N][N]));
-    if (!B) {
-        fprintf(stderr, "error allocating memory for matrix\n");
-        exit(1);
-    }
-    // fill array with numbers
+
     for (int i = 0; i < 4; i++) {
         for (int j = 0; j < 4; j++) {
-            A[i][j] = (float) count++;
+            if (fabs(A[i][j] - B[i][j]) > epsilon) {
+                are_equal = 0;
+                break;
+            }
         }
-    }
-    count = 0;
-    for (int i = 0; i < 4; i++) {
-        for (int j = 0; j < 4; j++) {
-            B[i][j] = (float) count++;
+        if (!are_equal) {
+            break;
         }
     }
 
-    struct timeval tv1, tv2, tv3, diff1, diff2;
-    gettimeofday(&tv1, NULL);
-
-    // normal scalar version
+    if (are_equal) {
+        printf("Arrays A and B are the same\n");
+    } else {
+        printf("Arrays A and B are not the same\n");
+    }
+}
+void
+scalarxmat44(float A[4][4], float lambda) {
     for (int i = 0; i < 4; i++) {
         for (int j = 0; j < 4; j++) {
-            A[i][j] = lamda * A[i][j];
+            A[i][j] = lambda * A[i][j];
         }
     }
-    gettimeofday(&tv2, NULL);
-
-    /*
-    // SSE version
-    __m128 lamda_vec = _mm_set1_ps(lamda);   // create a vector with 4 copies of lambda
+}
+void
+scalarxmat44SSE(float B[4][4], float lambda) {
```

Comment:
nitpicking, but could you please rename the functions as scalarxmat44_c() and scalarxmat44_sse()?